godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,875 @@
+        self.as_ref().class_name()
+    }
+
+    fn set(&mut self, name: StringName, value: &Variant) -> bool {
+        self.as_mut().set(name, value)
+    }
+
+    fn get(&self, name: StringName) -> Option<Variant> {
+        self.as_ref().get(name)
+    }
+
+    fn property_list(&self) -> &[PropertyInfo] {
+        self.as_ref().property_list()
+    }
+
+    fn method_list(&self) -> &[MethodInfo] {
+        self.as_ref().method_list()
+    }
+
+    fn call(
+        &mut self,
+        method: StringName,
+        args: &[&Variant],
+    ) -> Result<Variant, sys::GDExtensionCallErrorType> {
+        self.as_mut().call(method, args)
+    }
+
+    fn get_script(&self) -> &Gd<Script> {
+        self.as_ref().get_script()
+    }
+
+    fn is_placeholder(&self) -> bool {
+        self.as_ref().is_placeholder()
+    }
+
+    fn has_method(&self, method: StringName) -> bool {
+        self.as_ref().has_method(method)
+    }
+
+    fn get_property_type(&self, name: StringName) -> VariantType {
+        self.as_ref().get_property_type(name)
+    }
+
+    fn to_string(&self) -> GString {
+        self.as_ref().to_string()
+    }
+
+    fn owner(&self) -> Gd<Object> {
+        self.as_ref().owner()
+    }
+
+    fn property_state(&self) -> Vec<(StringName, Variant)> {
+        self.as_ref().property_state()
+    }
+
+    fn language(&self) -> Gd<ScriptLanguage> {
+        self.as_ref().language()
+    }
+
+    fn on_refcount_incremented(&self) {
+        self.as_ref().on_refcount_incremented();
+    }
+
+    fn on_refcount_decremented(&self) -> bool {
+        self.as_ref().on_refcount_decremented()
+    }
+
+    fn property_get_fallback(&self, name: StringName) -> Option<Variant> {
+        self.as_ref().property_get_fallback(name)
+    }
+
+    fn property_set_fallback(&mut self, name: StringName, value: &Variant) -> bool {
+        self.as_mut().property_set_fallback(name, value)
+    }
+}
+
+struct ScriptInstanceData<T: ScriptInstance> {
+    inner: RefCell<T>,
+    gd_instance_ptr: *mut sys::GDExtensionScriptInstanceInfo,
+}
+
+impl<T: ScriptInstance> Drop for ScriptInstanceData<T> {
+    fn drop(&mut self) {
+        // SAFETY: The ownership of ScriptInstaceData is transferred to Godot after it's creation. The engine then calls
+        // script_instance_info::free_func when it frees its own GDExtensionScriptInstance and subsequently wants to drop the ScriptInstanceData.
+        // After the the data has been dropped the instance info is no longer being used, but never freed. It is therefore safe to drop the
+        // instace info at the same time.
+        let instance = unsafe { Box::from_raw(self.gd_instance_ptr) };
+
+        drop(instance);
+    }
+}
+
+/// Creates a new [`sys::GDExtensionScriptInstancePtr`] from a type that implements [`ScriptInstance`].
+///
+/// This is the only way to obtain a new `GDExtensionScriptInstance`. See the [`ScriptInstance`] trait for more details.
+pub fn create_script_instance<T: ScriptInstance>(
```

Comment:
The first link doesn't work because it refers to a private symbol, see [rendered doc](https://godot-rust.github.io/docs/gdext/pr-492/godot/builtin/fn.create_script_instance.html). In fact, I would not mention `GDExtensionScriptInstancePtr` at all here.

As stated in the main review comment, I would probably return `*mut c_void` directly. A `//` comment could mention that this matches `sys::GDExtensionScriptInstancePtr`.